nullaway/src/test/java/com/uber/nullaway/NullAwayContractsBooleanTests.java

```diff
@@ -0,0 +1,408 @@
+            "      // BUG: Diagnostic contains: dereferenced expression",
+            "      && o1.toString().isEmpty();",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void contractUnreachablePath() {
+    helper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "class Test {",
+            "  String test(Object required) {",
+            "    return Validation.identity(required == null)",
+            "      ? required.toString()",
+            "      : required.toString();",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void contractUnreachablePathAfterFailure() {
+    helper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "class Test {",
+            "  String test(@Nullable Object o) {",
+            "    Validation.checkTrue(o == null);",
+            "    return Validation.identity(o == null)",
+            "      // BUG: Diagnostic contains: dereferenced expression",
+            "      ? o.toString()",
+            // This path is unreachable because o is guaranteed to be null
+            // after checkTrue(o == null). No failures should be reported.
+            "      : o.toString();",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void contractNestedBooleanNullness() {
+    helper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "class Test {",
+            "  String test(@Nullable Object o) {",
+            "    return Validation.identity(o == null)",
+            "      ? (Validation.identity(o != null)",
+            "        ? o.toString()",
+            "        // BUG: Diagnostic contains: dereferenced expression",
+            "        : o.toString())",
+            "      : (Validation.identity(o != null)",
+            "        ? o.toString()",
+            "        : o.toString());",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  private CompilationTestHelper helper() {
+    return makeTestHelperWithArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber"))
+        .addSourceLines(
+            "Validation.java",
+            "package com.uber;",
+            "import org.jetbrains.annotations.Contract;",
+            "public final class Validation {",
+            "  @Contract(\"false -> fail\")",
+            "  static void checkTrue(boolean value) {",
+            "    if (!value) throw new RuntimeException();",
+            "  }",
+            "  @Contract(\"true -> fail\")",
+            "  static void checkFalse(boolean value) {",
+            "    if (!value) throw new RuntimeException();",
```

Comment:
I think this should throw when `value == true`

```suggestion
            "    if (value) throw new RuntimeException();",
```

Reply:
You're absolutely right, thanks!